primeqa/ir/dense/colbert_top/colbert/search/index_loader.py

```diff
@@ -19,17 +21,20 @@ def __init__(self, index_path):
         self._load_embeddings()
 
     def _load_codec(self):
+        print_message(f"#> Loading codec...")
         self.codec = ResidualCodec.load(self.index_path)
 
     def _load_ivf(self):
-        ivf, ivf_lengths = torch.load(os.path.join(self.index_path, "ivf.pt"), map_location='cpu')
-        
-        if False:
-            ivf = ivf.tolist()
-            ivf = [ivf[offset:endpos] for offset, endpos in lengths2offsets(ivf_lengths)]
+        print_message(f"#> Loading IVF...")
+
+        if os.path.exists(os.path.join(self.index_path, "ivf.pid.pt")):
+            ivf, ivf_lengths = torch.load(os.path.join(self.index_path, "ivf.pid.pt"), map_location='cpu')
         else:
-            # ivf, ivf_lengths = ivf.cuda(), torch.LongTensor(ivf_lengths).cuda()  # FIXME: REMOVE THIS LINE!
-            ivf = StridedTensor(ivf, ivf_lengths)
+            assert os.path.exists(os.path.join(self.index_path, "ivf.pt"))
```

Comment:
It might be good to have messages in more of the asserts, e.g.
> assert os.path.exists(os.path.join(self.index_path, "ivf.pt")), f"ivf.pt not found in {self.index_path}"

Reply:
Good point, will add these